File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) 
    {
        unordered_set<int> set(nums.begin(),nums.end());
        ListNode* dummy=new ListNode(-1);
        ListNode *prev=dummy,*curr=head;
        dummy->next=head;
        while(curr)
        {
            if(set.count(curr->val))
            {
                prev->next=curr->next;
                delete curr;
                curr=prev->next;
            }
            else
            {
                prev=curr;
                curr=curr->next;
            }
        }
        return dummy->next;
    }
};

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) 
    {
        HashSet<Integer> set=new HashSet<>();
        for(int num : nums)
        set.add(num);

        ListNode dummyNode=new ListNode(-1);
        dummyNode.next=head;

        ListNode current=head,previous=dummyNode;
        while(current!=null)
        {
            if(set.contains(current.val))
            previous.next=current.next;
            else
            previous=current;

            current=current.next;
        }
        return dummyNode.next;
    }
}